src/kakarot/instructions/system_operations.cairo

```diff
@@ -72,16 +73,38 @@ namespace SystemOperations {
         let size = popped[2];
         let (_salt) = salt.read();
 
-        // create2 dynamic gas:
+        // create dynamic gas:
         // dynamic_gas = 6 * minimum_word_size + memory_expansion_cost + deployment_code_execution_cost + code_deposit_cost
         // -> ``memory_expansion_cost + deployment_code_execution_cost + code_deposit_cost`` is handled inside ``initialize_sub_context``
         let (minimum_word_size) = Helpers.minimum_word_count(size.low);
         let word_size_gas = 6 * minimum_word_size;
         let ctx = ExecutionContext.increment_gas_used(self=ctx, inc_value=word_size_gas);
 
+        let (evm_contract_address, starknet_contract_address) = ContractAccount.deploy_create(
+            _salt
+        );
+
+        let (bytecode: felt*) = alloc();
```

Comment:
since create2's address construction depends on bytecode, i refactored `CreateHelper.initialize_sub_context` to take bytecode as argument